device/tt_cluster_descriptor.cpp

```diff
@@ -523,15 +489,20 @@ void tt_ClusterDescriptor::load_ethernet_connections_from_connectivity_descripto
 }
 
 void tt_ClusterDescriptor::load_chips_from_connectivity_descriptor(YAML::Node &yaml, tt_ClusterDescriptor &desc) {
-    for (YAML::const_iterator node = yaml["chips"].begin(); node != yaml["chips"].end(); ++node) {
+    auto chip_coordinate_map = yaml["chips"].as<std::map<int, std::vector<int>>>();
+    desc.chips_have_coordinates = not chip_coordinate_map.empty();
+
+    for (YAML::const_iterator node = yaml["arch"].begin(); node != yaml["arch"].end(); ++node) {
         chip_id_t chip_id = node->first.as<int>();
-        std::vector<int> chip_rack_coords = node->second.as<std::vector<int>>();
-        log_assert(chip_rack_coords.size() == 4, "Galaxy (x, y, rack, shelf) coords must be size 4");
-        eth_coord_t chip_location{
-            chip_rack_coords.at(0), chip_rack_coords.at(1), chip_rack_coords.at(2), chip_rack_coords.at(3)};
-        
-        desc.chip_locations.insert({chip_id, chip_location});
-        desc.coords_to_chip_ids[std::get<2>(chip_location)][std::get<3>(chip_location)][std::get<1>(chip_location)][std::get<0>(chip_location)] = chip_id;
+        // Not all archs (e.g. BH) have chip coordinates
```

Comment:
nit: e.g. BH during bringup and GS

Reply:
Correct me if I am wrong @TTDRosen  but I don't think BH will have chip coordinates even after bring-up